TACHO_DRIVER.md

```diff
@@ -0,0 +1,199 @@
+| ------------ | -----------
+| 0            | Operation sucessful
+| ( -1) EPERM  | Operation not permitted (not the same as EACCESS!) 
+| (-13) EACCES | Permission denied
+| (-13) EBUSY  | Device or resource busy
+| (-22) EINVAL | Invalid argument
+
+The attributes operate as described in `spec.json` - with the following
+caveats for the `speed_regulation` value:
+
+| `speed_regulation` | Write `duty_cycle_sp` | Write `speed_sp`
+| ------------------ | --------------------- | ----------------
+| `off`              | Works                 | Return EPERM
+| `on`               | Return EPERM          | Works
+
+The `ramp_up_sp` and `ramp_down_sp` specify how long (in msec) the motor
+will take to go from 0 to 100 of the controlling parameter depending on
+the `speed_regulation` mode.
+
+| `speed_regulation` | Ramp control parameter
+| ------------------ | ----------------------
+| `off`              | 0 to 100% Duty Cycle                 
+| `on`               | 0 to 100% Max Speed
+
+The maximum speed of the motor is set in the driver. The large motor has a
+maximum speed of 1200 pulses per second and the medium motor has a maximum
+speed of 900 pulses per second. Both motors have 360 pulses per revolution.
+
+## Ramping
+
+See `spec.json` for a description of the `ramp_up_sp` and `ramp_down_sp`
+attributes. Except in `run-direct` mode, you cannot change the ramping
+attributes while the motor is running.
+
+If you change the allowed `duty_cycle_sp` or `speed_sp` value while the
+motor is in `run_direct` mode, the ramp time setpoints govern the operation.
+
+No matter the sign of the `duty_cycle_sp` or `speed_sp`, if the new
+setpoint is towards a lower absolute value, then the `ramp_down_sp`
+determines how the motor slows down. If the new setpoint is towards a
+higher absolute value, then the `ramp_up_sp` determines how the motor
+speeds up.
+
+An example follows:
+
+- `ramp_up_sp` is set to 4000
+- `ramp_down_sp` is set to 2000
+- `speed_regulation` is set to `off`
+- `duty_cycle_sp` is set to 75%
+
+If we write `run-direct` to the motor `command` attribute, the motor will
+take 3000 msec to ramp up to 75% duty cycle. What happens when we change the
+`duty_cycle_sp` to other values while the motor is running?
+
+| New `duty_cycle_sp` | Operation
+| ------------------- | ----------------------
+|   100               | Motor ramps up to 100% in 1000 msec
+|   50                | Motor ramps down to 50% in 1000 msec
+|   -50               | Motor ramps down to 0% in 1000 msec and up to -50 in 2000 msec
+
+## Run Modes
+
+The `run` attribute determines function that the motor is executing. Within a `run`
+mode, some attributes have an effect, while others do not. These details are
+described below.
+
+The run modes can be classified into two types. The `run-direct` is suited to
+applications where the programmer wants direct control of the motors. The other
+three modes:
+
+- `run_timed`
+- `run_to_abs_pos`
+- `run_to_rel_pos`
```

Comment:
Here is a suggestion for `run-direct` with `speed_sp`. Instead of making `run-direct` have multiple modes of operation, let's add more `run-direct` commands. Some users have expected something like `run-direct` with `position_sp` as well, so why not throw that into the mix.

Proposal:
* Rename `run-direct` to `run-direct-duty`. This will work as it currently does. Responds to changes in `duty_cycle_sp`. Ignores all other attributes. Having `duty` in the command name should also help with expectations that `speed_sp` wont' work here.
* Add new command `run-direct-speed`. This command responds to changes in `speed_sp` and ignores all other attributes.
* Add new command `run-direct-pos`. This command responds to changes in `position_sp` and ignores all other attributes. (We might need to look at speed for this one. If we do, I think speed should work like non-direct commands and use what was set when the command was sent).

---

And to answer your question, yes, `stop` counts as a command.